File: Lista nr 3/Square.cpp
#include "Square.hpp"
#include "Quadrangle.hpp"
#include "NegativeNumber.hpp"
Square::Square(int a)
{
    if(a < 0)
        throw NegativeNumber();
    side1 = a;
}

int Square::area()
{
    return side1 * side1;
}
int Square::perimeter()
{
    return 4*side1;
}
std::string Square::getNAME()
{
    return "Square";
}

Square::~Square()
{

}

File: Lista nr 3 Java/Square.java
public class Square extends Quadrangle
{
    public Square(int a) throws NegativeNumber
    {
        if(a < 0)
            throw new NegativeNumber();
        side1 = a;
    }
    public int area()
    {
        return side1 * side1;
    }
    public int perimeter()
    {
        return 4*side1;
    }
    public String getNAME()
    {
        return "Square";
    }
}
